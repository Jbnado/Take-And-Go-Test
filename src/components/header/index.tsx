import {
  House,
  ChevronDown,
  Bell,
  Gear,
  PersonFill,
} from "react-bootstrap-icons";

export function Header() {
  return (
    <header aria-label="Page Header" className="bg-white shadow py-4 px-5">
      <div className="flex items-center sm:justify-between gap-4">
        <div className="flex gap-4 items-center">
          <img src="/Logo.png" alt="Logo" className="h-8 w-auto" />
          <div className="sm:flex hidden h-5 w-5 group shrink-0 items-center justify-center rounded-full transition text-blue-600 border-2 border-blue-600">
            <House size={14} />
          </div>
          <div className="sm:flex hidden gap-2 items-center cursor-pointer">
            <span className="font-medium">Dryve - Ribeirão Preto</span>
            <ChevronDown size={12} />
          </div>
        </div>
        <div className="flex sm:gap-7 gap-5 items-center">
          <a
            href="#"
            className="block shrink-0 text-gray-600 hover:text-gray-700"
          >
            <span className="sr-only">Notificações</span>
            <Bell size={16} />
          </a>
          <a
            href="#"
            className="block shrink-0 text-gray-600 hover:text-gray-700"
          >
            <span className="sr-only">Configurações</span>
            <Gear size={16} />
          </a>
          <button
            type="button"
            className="p-1 group flex shrink-0 items-center justify-center rounded-full transition text-blue-400 shadow-sm hover:text-blue-500 border-2 border-blue-600"
          >
            <div className="rounded-full bg-blue-200 p-1">
              <span className="sr-only">Usuário</span>
              <PersonFill size={20} />
            </div>
          </button>
        </div>
      </div>
    </header>
  );
}
