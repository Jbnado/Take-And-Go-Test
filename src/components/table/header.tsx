import { useCallback, useRef } from "react";
import { Plus, Search, Sliders } from "react-bootstrap-icons";

export function TableHeader({
  title,
  onSearch,
}: {
  title: string;
  onSearch?: (value: string) => void;
}) {
  const inputValue = useRef<HTMLInputElement | null>(null);

  const handleSearch = useCallback(() => {
    onSearch?.(inputValue.current?.value as string);
  }, [onSearch, inputValue]);

  const handleSearchButtonClick = useCallback(() => {
    handleSearch();
  }, [handleSearch]);

  const handleSearchKeyDown = useCallback(
    (event: React.KeyboardEvent<HTMLInputElement>) => {
      if (event.key === "Enter") {
        event.preventDefault();
        handleSearch();
      }
    },
    [handleSearch]
  );

  return (
    <header className="flex flex-col lg:flex-row lg:items-center lg:justify-between gap-4 h-50 mb-4">
      <div className="flex gap-4 items-center">
        <h1 className="text-2xl font-semibold">{title}</h1>
        {onSearch ? (
          <div className="relative sm:block hidden">
            <label className="sr-only" htmlFor="search"></label>
            <input
              className="h-10 w-full rounded-full border-none bg-white pe-10 ps-4 text-sm shadow-sm sm:w-56"
              id="search"
              type="text"
              placeholder="Buscar por veÃ­culo"
              ref={inputValue}
              onKeyDown={handleSearchKeyDown}
            />
            <button
              type="button"
              onClick={handleSearchButtonClick}
              className="absolute end-1 top-1/2 -translate-y-1/2 rounded-full bg-gray-50 p-2 text-gray-600 transition hover:text-gray-700"
            >
              <span className="sr-only">Search</span>
              <Search />
            </button>
          </div>
        ) : null}
      </div>
      <div className="flex gap-4 items-center">
        <button className="flex gap-2 items-center rounded border border-blue-600 bg-white sm:px-6 px-4 py-2 text-sm font-medium text-blue-600 hover:bg-blue-600 hover:text-white focus:outline-none focus:ring active:text-white">
          <Sliders size={18} />
          Filtrar
        </button>
        <button className="flex gap-2 items-center rounded border border-blue-600 bg-blue-600 sm:px-6 px-4 py-2 text-sm font-medium text-white hover:bg-transparent hover:text-blue-600 focus:outline-none focus:ring active:text-blue-600">
          <Plus size={20} />
          Adicionar
        </button>
      </div>
    </header>
  );
}
