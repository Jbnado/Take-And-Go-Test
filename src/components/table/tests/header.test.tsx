import TestRenderer from "react-test-renderer";
import { fn } from "jest-mock";
import { describe, it, expect } from "vitest";
import { TableHeader } from "../header";

describe("TableHeader Component", () => {
  it("renders correctly with title", () => {
    const tree = TestRenderer.create(
      <TableHeader title="Test Title" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with search input", () => {
    const tree = TestRenderer.create(
      <TableHeader title="Test Title" onSearch={fn()} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
