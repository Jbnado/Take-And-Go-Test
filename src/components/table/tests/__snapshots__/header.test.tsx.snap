// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableHeader Component > renders correctly with search input 1`] = `
<header
  className="flex flex-col lg:flex-row lg:items-center lg:justify-between gap-4 h-50 mb-4"
>
  <div
    className="flex gap-4 items-center"
  >
    <h1
      className="text-2xl font-semibold"
    >
      Test Title
    </h1>
    <div
      className="relative sm:block hidden"
    >
      <label
        className="sr-only"
        htmlFor="search"
      />
      <input
        className="h-10 w-full rounded-full border-none bg-white pe-10 ps-4 text-sm shadow-sm sm:w-56"
        id="search"
        onKeyDown={[Function]}
        placeholder="Buscar por veÃ­culo"
        type="text"
      />
      <button
        className="absolute end-1 top-1/2 -translate-y-1/2 rounded-full bg-gray-50 p-2 text-gray-600 transition hover:text-gray-700"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Search
        </span>
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="flex gap-4 items-center"
  >
    <button
      className="flex gap-2 items-center rounded border border-blue-600 bg-white sm:px-6 px-4 py-2 text-sm font-medium text-blue-600 hover:bg-blue-600 hover:text-white focus:outline-none focus:ring active:text-white"
    >
      <svg
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.5 2a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM9.05 3a2.5 2.5 0 0 1 4.9 0H16v1h-2.05a2.5 2.5 0 0 1-4.9 0H0V3h9.05zM4.5 7a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM2.05 8a2.5 2.5 0 0 1 4.9 0H16v1H6.95a2.5 2.5 0 0 1-4.9 0H0V8h2.05zm9.45 4a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zm-2.45 1a2.5 2.5 0 0 1 4.9 0H16v1h-2.05a2.5 2.5 0 0 1-4.9 0H0v-1h9.05z"
          fillRule="evenodd"
        />
      </svg>
      Filtrar
    </button>
    <button
      className="flex gap-2 items-center rounded border border-blue-600 bg-blue-600 sm:px-6 px-4 py-2 text-sm font-medium text-white hover:bg-transparent hover:text-blue-600 focus:outline-none focus:ring active:text-blue-600"
    >
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 16 16"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
        />
      </svg>
      Adicionar
    </button>
  </div>
</header>
`;

exports[`TableHeader Component > renders correctly with title 1`] = `
<header
  className="flex flex-col lg:flex-row lg:items-center lg:justify-between gap-4 h-50 mb-4"
>
  <div
    className="flex gap-4 items-center"
  >
    <h1
      className="text-2xl font-semibold"
    >
      Test Title
    </h1>
  </div>
  <div
    className="flex gap-4 items-center"
  >
    <button
      className="flex gap-2 items-center rounded border border-blue-600 bg-white sm:px-6 px-4 py-2 text-sm font-medium text-blue-600 hover:bg-blue-600 hover:text-white focus:outline-none focus:ring active:text-white"
    >
      <svg
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.5 2a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM9.05 3a2.5 2.5 0 0 1 4.9 0H16v1h-2.05a2.5 2.5 0 0 1-4.9 0H0V3h9.05zM4.5 7a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM2.05 8a2.5 2.5 0 0 1 4.9 0H16v1H6.95a2.5 2.5 0 0 1-4.9 0H0V8h2.05zm9.45 4a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zm-2.45 1a2.5 2.5 0 0 1 4.9 0H16v1h-2.05a2.5 2.5 0 0 1-4.9 0H0v-1h9.05z"
          fillRule="evenodd"
        />
      </svg>
      Filtrar
    </button>
    <button
      className="flex gap-2 items-center rounded border border-blue-600 bg-blue-600 sm:px-6 px-4 py-2 text-sm font-medium text-white hover:bg-transparent hover:text-blue-600 focus:outline-none focus:ring active:text-blue-600"
    >
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 16 16"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
        />
      </svg>
      Adicionar
    </button>
  </div>
</header>
`;
